wallet/src/main/java/com/android/identity_credential/wallet/util/ActivityLogger.kt

```diff
@@ -0,0 +1,307 @@
+import com.android.identity.util.UUID
+import com.android.identity_credential.wallet.util.ActivityLogger.getString
+import org.json.JSONException
+import org.json.JSONObject
+
+/**
+ * Activity Logging facility.
+ *
+ * This saves activities in Cbor format using storageEngine.
+ */
+
+private const val TAG = "ActivityLogger"
+
+enum class ActivityLogging {
+    PRESENTATION, COMMUNICATION
+}
+
+enum class PresentationType {
+    OpenID4VP, CREDMAN
+}
+
+private var activityLoggingEnabled = false
+
+sealed class ActivityEntry(
+    open var activityId: Long = ActivityLogger.generateUniqueID(),
+    open var documentId: String = "",
+    open var timestamp: Long = System.currentTimeMillis(),
+    open var deviceRequest: String?,
+    open var deviceResponse: String?,
+    open var type: ActivityLogging,
+) {
+    data class PresentationActivity(
+        override var activityId: Long = ActivityLogger.generateUniqueID(),
+        override var documentId: String = "",
+        override var timestamp: Long = System.currentTimeMillis(),
+        var presentationType: PresentationType? = null,
+        var requestedFields: Array<String>? = null,
+        override var deviceRequest: String? = "",
+        override var deviceResponse: String? = "",
+        override var type: ActivityLogging = ActivityLogging.PRESENTATION
+    ) : ActivityEntry(activityId, documentId, timestamp, deviceRequest, deviceResponse, type)
+
+    data class CommunicationActivity(
+        override var activityId: Long = ActivityLogger.generateUniqueID(),
+        override var documentId: String = "",
+        override var timestamp: Long = System.currentTimeMillis(),
+        var url: String? = "",
+        var requestType: String? = null,
+        override var deviceRequest: String? = "",
+        override var deviceResponse: String? = "",
+        override var type: ActivityLogging = ActivityLogging.COMMUNICATION
+    ) : ActivityEntry(activityId, documentId, timestamp, deviceRequest, deviceResponse, type)
+}
+
+class PresentationActivityBuilder {
+    private var activityId: Long = ActivityLogger.generateUniqueID()
+    private var documentId: String = ""
+    private var timestamp: Long = System.currentTimeMillis()
+    private var presentationType: PresentationType? = null
+    private var requestedFields: Array<String>? = null
+    private var deviceRequest: String? = null
+    private var deviceResponse: String? = null
+    private var type: ActivityLogging = ActivityLogging.PRESENTATION
+
+    fun setActivityId(activityId: Long) = apply { this.activityId = activityId }
+    fun setDocumentId(documentId: String) = apply { this.documentId = documentId }
+    fun setTimestamp(timestamp: Long) = apply { this.timestamp = timestamp }
+    fun setPresentationType(presentationType: PresentationType) = apply { this.presentationType = presentationType }
+    fun setRequestedFields(requestedFields: Array<String>) = apply { this.requestedFields = requestedFields }
```

Comment:
The requested fields are already part of the DeviceRequest CBOR that we are storing... so I don't think we need to extra that and store it alongside.

I gather you do this b/c you want them easily available when viewing the event but I think the right approach is to have the Model associated with the log viewer take the event, decode DeviceRequest - using [DeviceRequestParser](https://github.com/openwallet-foundation-labs/identity-credential/blob/main/identity-mdoc/src/commonMain/kotlin/com/android/identity/mdoc/request/DeviceRequestParser.kt) - and build up some simple data classes that the Compose code can output. Kinda similar to what we do with `ReaderModel` / `ReaderScreen` and `DocumentModel` / `DocumentDetailsScreen`. See comment on file `ActivityLogScreen.kt` for more context.

Reply:
Done.